### PR TITLE
Validate LOCODE attributes of candidates to the network map

### LINKED ISSUE #316: Use UN/LOCODE as location identity in storage node attributes
NeoFS storage nodes specify their location in node attributes. These attributes presented in the network map, so one can define placement policies with location attributes from there. However there are issues:
- attributes do not have strict formatting rules for location,
- there is more than one way you can name location, e.g. `St. Petersburg` or `Saint Petersburg`. 

To deal with these issues we can use [UN/LOCODE](https://unece.org/unlocode) as location attribute. LOCODEs are `XX XXX` ids, e.g. `CN SGH` for Shanghai. They stored in a database maintained by UN in this format:
```
CN SGH    Shanghai    Shanghai    SH    12345---    AS    1401    3114N 12129E
```

## Example

1. Storage node won't be included to the network map without LOCODE. So at first the user looks up through LOCODE database or use external service (maybe `neofs-cli` helper command?) to find proper LOCODE value.
2. User sets LOCODE in storage node config (`NEOFS_NODE_LOCODE=RU LED`) and then starts it.
3. Storage node invokes `AddPeer` method with storage node attribute `/LOCODE:RU LED`
4. Inner ring nodes makes sanity check of the storage node and converts this LOCODE to `/Country:Russia/City:Saint Petersburg`
5. Inner ring nodes approve storage node with parsed LOCODE and it gets into network map this way.

## Issues
- LOCODE encodes country, city or big transport hubs within it such as airports, railway station, etc. However it does not define regions like `Europe` or `Asia`.

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #316.

### CODE REVIEW

File: pkg/innerring/processors/netmap/nodevalidation/locode/validator.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// the Validator is immediately ready to work through API.
+type Validator struct {
+	db DB
+
+	mAttr map[string]attrDescriptor
+}
+
+// TODO: define constants in API lib.
+const (
+	attrKeyLocode = "Locode"
```
Comment (alexvanin) [inline pkg/innerring/processors/netmap/nodevalidation/locode/validator.go:31]: Most docs I've seen spell `LOCODE` or `UN/LOCODE` in all caps. I think storage node attribute should look the same: `/LOCODE:RU LED`

Thoughts? @cthulhu-rider @realloc

**Review by alexvanin — CHANGES_REQUESTED** [review]
Close #316 when storage node will require LOCODE attribute (e.g. from config). You can do that in separate PR.

File: pkg/innerring/processors/netmap/nodevalidation/locode/validator.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/innerring/processors/netmap/nodevalidation/locode/validator.go:31]: I just followed current API [constant](https://github.com/nspcc-dev/neofs-api/blob/113c7d97d1951d140c26509000b6ff8df7d1ec64/netmap/types.proto#L153). It seems like nothing is blocking to make an uppercase constant.

File: pkg/innerring/locode.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	irlocode "github.com/nspcc-dev/neofs-node/pkg/innerring/processors/netmap/nodevalidation/locode"
+	"github.com/nspcc-dev/neofs-node/pkg/util/locode"
+	locodedb "github.com/nspcc-dev/neofs-node/pkg/util/locode/db"
+	locodebolt "github.com/nspcc-dev/neofs-node/pkg/util/locode/db/boltdb"
+	"github.com/spf13/viper"
+)
+
+func (s *Server) newLocodeValidator(cfg *viper.Viper) (netmap.NodeValidator, error) {
+	locodeDB := locodebolt.New(locodebolt.Prm{
+		Path: cfg.GetString("locode.db.path"),
```
[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply pkg/innerring/locode.go:14]: You mean define exported constant or set the default value?

[the remark this answers is not in the document]
Comment (alexvanin) [reply pkg/innerring/locode.go:14]: Set default value (empty).

[the remark this answers is not in the document]
Comment (cthulhu-rider) [reply pkg/innerring/locode.go:14]: Done

**Comment by alexvanin** [comment]
After discussion we've decided to make separate config records for `UN-LOCODE` attribute. We'll do that later in separate issue: https://github.com/nspcc-dev/neofs-node/issues/376

However for now we should treat `UN-LOCODE` as well-known attribute in storage node. Make corresponding changes at `cmd/neofs-node/attributes.go` in this PR.

/cc @cthulhu-rider @realloc

File: pkg/innerring/processors/netmap/nodevalidation/locode/validator.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/innerring/processors/netmap/nodevalidation/locode/validator.go:31]: We decided with @realloc  and @alexvanin to rename `Locode` -> `UN-LOCODE`.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**